runtime/doc/lsp.txt

```diff
@@ -50,7 +50,7 @@ listed below, if (1) the language server supports the functionality and (2)
 the options are empty or were set by the builtin runtime (ftplugin) files. The
 options are not restored when the LSP client is stopped or detached.
 
-- 'omnifunc' is set to |vim.lsp.omnifunc()|, use |i_CTRL-X_CTRL-O| to trigger
+- 'omnifunc' is set to |vim.lsp.completion.omnifunc()|, use |i_CTRL-X_CTRL-O| to trigger
```

Comment:
Not sure if renaming this is a breaking change. @mfussenegger do you think we should keep this guy in the parent `lsp` module instead of having it inside the new `completion` one?

Reply:
It is breaking, yes, so maybe keep a shim with deprecation notice in the parent module (while leaving the implementation where you put it)? If we do punt this after 0.10, this'll mean we'll have to live with it for a bit, though...

Reply:
What is the rationale for renaming/deprecating this? It seems like needless churn for no benefit. Since `omnifunc` corresponds to an option name, the fact that it’s part of the `completion` module is an implementation detail (just as we also have `vim.lsp.tagfunc`).

I won’t die on this hill but my preference would be to leave this as-is, without deprecating, especially since there doesn’t seem to be a compelling reason to move it.

Reply:
I don't really mind what we do with this. @justinmk what are your thoughts in terms of codebase conventions/organization?

Reply:
To get this PR over the finish line we could keep `vim.lsp.omnifunc` undeprecated for now and rename `vim.lsp.completion.omnifunc` to `vim.lsp.completion._omnifunc` + `@nodoc`.

We can then follow up on this later - depending on the direction we want to go with it.

Reply:
To be clear, the load-bearing part of my comment was "keep a shim", _not_ the deprecation message. I'm fully on-board with just moving the _implementation_, not the exposure point.